Golem/Factory.cs

```diff
@@ -19,7 +25,7 @@ public async Task<IGolem> Create(string modulesDir, ILoggerFactory? loggerFactor
 
         public async Task<IGolem> Create(string modulesDir, ILoggerFactory? loggerFactory = null, bool mainnet = true, string? dataDir = null)
         {
-            return await Create(modulesDir, loggerFactory, mainnet, null, RelayType.Public);
+            return await Create(modulesDir, loggerFactory, mainnet, dataDir, DecideRelayType());
```

Comment:
It seems error prone that the override by env is used in one function only. If the user will use the env variable to set the type but at some point in the future the function used in code will be change to another one that doesn't allow for this override then there will be no warning about the change of behavior.